server/errorstore/errors.go

```diff
@@ -87,33 +94,46 @@ func runHandler(ctx context.Context, eh *Handler) {
 //
 // If flush is `true`, performs a destructive read - the errors are removed
 // from Redis on return.
-func (h *Handler) Retrieve(flush bool) ([]string, error) {
-	errorKeys, err := redis.ScanKeys(h.pool, "error:*", 100)
+func (h *Handler) Retrieve(flush bool) ([]*storedError, error) {
+	// scanning only the error:*:json keys as json and count are both tagged keys
+	// and should hash to the same Redis slot
+	errorKeys, err := redis.ScanKeys(h.pool, "error:*:json", 100)
 	if err != nil {
 		return nil, err
 	}
 
 	keysBySlot := redis.SplitKeysBySlot(h.pool, errorKeys...)
-	var errors []string
+	var rawErrs []interface{}
 	for _, qkeys := range keysBySlot {
 		if len(qkeys) > 0 {
 			gotErrors, err := h.collectBatchErrors(qkeys, flush)
 			if err != nil {
 				return nil, err
 			}
-			errors = append(errors, gotErrors...)
+			rawErrs = append(rawErrs, gotErrors...)
 		}
 	}
-	return errors, nil
+
+	errorList := []*storedError{}
```

Comment:
Nit, but you can allocate the required space immediately, potentially saving some subsequent allocations to grow the slice:

```suggestion
	errorList := make([]*storedError, 0, len(rawErrs))
```

Reply:
really good catch, thanks!